hivemind/utils/networking.py

```diff
@@ -1,12 +1,16 @@
 import socket
 from contextlib import closing
 from ipaddress import ip_address
-from typing import Optional, Sequence
+from typing import Optional, Sequence, Tuple
 
 from multiaddr import Multiaddr
 
 Hostname, Port = str, int  # flavour types
-Endpoint = str  # e.g. 1.2.3.4:1337 or [2a21:6с8:b192:2105]:8888, https://networkengineering.stackexchange.com/a/9435
+Endpoint = (
+    Tuple[  # e.g. 1.2.3.4:1337 or [2a21:6с8:b192:2105]:8888, https://networkengineering.stackexchange.com/a/9435
+        str, Tuple[str, ...]
+    ]
+)
```

Comment:
```suggestion
```

Ideally, we should always use `PeerID` instead of `Endpoint`, so libp2p will choose a route to the target peer by itself (as I explained during our call).

If we postpone this change to the next PR, we still don't such an `Endpoint` tuple since it duplicates the existing `hivemind.PeerInfo` structure (but worse since it doesn't have the named fields).

Also, the comment here became irrelevant.

Reply:
What is your suggestion on `Endpoint`? I think we can delete it, because now we have `PeerInfo`, in future we want to have only `PeerID`. Both are already exiting structures

Reply:
Yeah, I suggest to delete it and use `PeerInfo` instead :) Replacing `PeerInfo` to `PeerID` can be done in a future PR (I can also do it myself, so you don't get overwhelmed) :)

Reply:
UPD: returned `utils/networking.py` back. Turned out it should be done more carefully, if should be at all.

Reply:
Good catch! Let's remove it along with all the grpc stuff, we can do this in a separate PR :)